activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -56,11 +56,16 @@ def concise_options(options)
     # +columns+ attribute of said TableDefinition object, in order to be used
     # for generating a number of table creation or table changing SQL statements.
     ColumnDefinition = Struct.new(:name, :type, :options, :sql_type) do # :nodoc:
+      self::OPTIONS = %i(limit precision scale default null collation comment
+                         primary_key).freeze
+      self::ADAPTER_OPTIONS = %i(array using cast_as auto_increment first after
```

Comment:
Where is this list common from?

Reply:
I looked through the different connection adapters(myql, postgres, etc) schema_definitions.rb along with test files to gather them.

Reply:
I'm not sure every adapter supports all options, but this is better than not checking at all. 🙂

Reply:
Can other adapters add items to this list? Could the list be inferred at runtime in some way rather than being hard coded?

https://github.com/rsim/oracle-enhanced (and other 3rd party adapters) may support other options and so this could break it.

Reply:
Good point @ghiculescu. For allowed column types we check the adapter (https://github.com/rails/rails/pull/42311) but I'm not sure we can do it for options 🤔